File: Homework/HW11/HW11.java
package Homework.HW11;

public class HW11 {
    public void floodFillJava(int[][] image, int sr, int sc, int newColor) {
        int originalColor = image[sr][sc];
        if (originalColor == newColor)
            return;

        floodFill(image, sr, sc, originalColor, newColor);
    }

    private void floodFill(int[][] image, int sr, int sc, int originalColor, int newColor) {
        if (sr < 0 || sr >= image.length || sc < 0 || sc >= image[0].length || image[sr][sc] != originalColor) {
            return;
        }

        image[sr][sc] = newColor;
        floodFill(image, sr + 1, sc, originalColor, newColor);
        floodFill(image, sr - 1, sc, originalColor, newColor);
        floodFill(image, sr, sc + 1, originalColor, newColor);
        floodFill(image, sr, sc - 1, originalColor, newColor);
    }

}

File: Homework/HW11/HW11.py
class Week11:

    def flood_fill_python(self, image, sr, sc, new_color):
        original_color = image[sr][sc]
        if original_color == new_color:
            return image

        def fill(r, c):
            if (0 <= r < len(image)) and (0 <= c < len(image[0])) and (image[r][c] == original_color):
                image[r][c] = new_color
                fill(r + 1, c)
                fill(r - 1, c)
                fill(r, c + 1)
                fill(r, c - 1)

        fill(sr, sc)
        return image
